pkg/v3/types/basetypes.go

```diff
@@ -216,12 +234,52 @@ func (r CheckResult) UniqueID() string {
 
 func (r CheckResult) String() string {
 	return fmt.Sprintf(
-		checkResultStringTemplate, r.PipelineExecutionState, r.Retryable, r.RetryInterval, r.Eligible,
+		checkResultStringTemplate, r.PipelineExecutionState, r.Retryable, r.Eligible,
 		r.IneligibilityReason, r.UpkeepID, r.Trigger, r.WorkID, r.GasAllocated,
 		hex.EncodeToString(r.PerformData), r.FastGasWei, r.LinkNative,
 	)
 }
 
+func (r CheckResult) MarshalJSON() ([]byte, error) {
+	crm := &checkResultMsg{
+		PipelineExecutionState: r.PipelineExecutionState,
+		Retryable:              r.Retryable,
+		Eligible:               r.Eligible,
+		IneligibilityReason:    r.IneligibilityReason,
+		UpkeepID:               r.UpkeepID,
+		Trigger:                r.Trigger,
+		WorkID:                 r.WorkID,
+		GasAllocated:           r.GasAllocated,
+		PerformData:            r.PerformData,
+		FastGasWei:             r.FastGasWei,
+		LinkNative:             r.LinkNative,
+	}
+
+	return json.Marshal(crm)
+}
+
+func (r *CheckResult) UnmarshalJSON(data []byte) error {
+	var crm checkResultMsg
+
+	if err := json.Unmarshal(data, &crm); err != nil {
+		return err
```

Comment:
can we add a temporary fix here which tries to unmarshall with an extra `RetryInterval` field for temporarily allowing compatibility with older versions

Reply:
thinking more, it will help in parsing observations from old versions, however it won't help in outcome consensus since that is done on raw bytes

Reply:
In case older versions are running in parallel, they'll expect to have the `RetryInterval` so even if newer versions will read that property they will need to include it when producing outcome/report